packages/fhir-location-management/src/components/LocationForm/index.tsx

```diff
@@ -155,6 +155,13 @@ const LocationForm = (props: LocationFormProps) => {
               const successUrl = successURLGenerator(payload);
               sendSuccessNotification(successMessage);
               afterSubmit?.(payload);
+              // hierarchy request usually takes quite a while to resolve, this coupled with react-query's request
+              // de-duping mechanism means that more recent requests will get deduped on the pending request.
+              // the pending request then resolves with stale data.
+              // This cancels any pending request so that after invalidation we can get a fresh promise launched then.
+              queryClient.cancelQueries([locationHierarchyResourceType]).catch((err) => {
```

Comment:
Does this work like AbortController().abort()? I see we are using an abort signal but not abort controller? Am I missing something?

Reply:
This implies so: https://react-query-v3.tanstack.com/guides/query-cancellation
> React Query provides each query function with an [AbortSignal instance](https://developer.mozilla.org/docs/Web/API/AbortSignal), if it's available in your runtime environment. When a query becomes out-of-date or inactive, this signal will become aborted. This means that all queries are cancellable